suite-common/suite-sync-quota-manager/src/tests/ensureDeviceHasQuotaThunk.test.ts

```diff
@@ -0,0 +1,215 @@
+import { combineReducers } from '@reduxjs/toolkit';
+
+import {
+    getProofOfDelegatedIdentity,
+    getPublicIdentityKeyFromDelegatedKey,
+} from '@suite-common/delegated-identity-key';
+import { DelegatedIdentityKey, TrezorDeviceWithState } from '@suite-common/suite-types';
+import { configureMockStore } from '@suite-common/test-utils';
+import TrezorConnect, { asProofOfDelegatedIdentity } from '@trezor/connect';
+import { err, ok } from '@trezor/type-utils';
+
+import { prepareChallengeSession } from '../challenge/prepareChallengeSession';
+import { DEFAULT_DEVICE_SIZE_QUOTA } from '../constants';
+import { ensureDeviceHasQuotaThunk } from '../ensureDeviceHasQuotaThunk';
+import {
+    SuiteSyncQuotaManagerState,
+    quotaManagerInitialState,
+    suiteSyncQuotaManagerReducer,
+} from '../quotaManagerReducer';
+import { checkStorageByPublicKey } from '../storage/checkStorage';
+import { registerStorageThunk } from '../storage/registerStorageThunk';
+
+jest.mock('../challenge/prepareChallengeSession', () => ({
+    prepareChallengeSession: jest.fn(),
+}));
+
+jest.mock('../storage/checkStorage', () => ({
+    checkStorageByPublicKey: jest.fn(),
+}));
+
+jest.mock('../storage/registerStorageThunk', () => ({
+    registerStorageThunk: jest.fn(),
+}));
+
+jest.mock('@suite-common/delegated-identity-key', () => ({
+    getProofOfDelegatedIdentity: jest.fn(),
+    getPublicIdentityKeyFromDelegatedKey: jest.fn(),
+}));
+
+jest.mock(
+    '@trezor/connect-analytics',
+    () => ({
+        EventType: {
+            DeviceSelected: 'DeviceSelected',
+        },
+    }),
+    { virtual: true },
+);
+
+const createStore = (statePatch?: Partial<SuiteSyncQuotaManagerState>) =>
+    configureMockStore({
+        reducer: combineReducers({
+            suiteSyncQuotaManager: suiteSyncQuotaManagerReducer,
+        }),
+        preloadedState: {
+            suiteSyncQuotaManager: {
+                ...quotaManagerInitialState,
+                baseUrl: 'https://quota-manager.test',
+                ...statePatch,
+            },
+        },
+    });
+
+const delegatedKey = 'deadbeefcafebabe' as unknown as DelegatedIdentityKey;
+const device = {
+    id: 'device-id',
+    features: { internal_model: 'T2T1' },
+} as unknown as TrezorDeviceWithState;
```

Comment:
pls no partials.
getSuiteDevice currently returns `TrezorDevice`, but maybe it could return `TrezorDeviceWithState`? 🤔 Would have to take a closer look.

```suggestion
import { getSuiteDevice } from '@suite-common/test-utils';
import { DeviceModelInternal } from '@trezor/device-utils';
const device = getSuiteDevice(
    { id: 'device-id' },
    { internal_model: DeviceModelInternal.T2T1 },
) as TrezorDeviceWithState;
```

Reply:
You dont need whole device

If you change signature to `device: Pick<TrezorDeviceWithState, 'id' | 'features'>;` you make your testing a bit easier